nemo/collections/nlp/data/language_modeling/megatron/retro_dataset.py

```diff
@@ -12,458 +12,199 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""RETRO Style dataset."""
+"""GPT style dataset."""
 
 import os
-from typing import List
+import time
 
 import numpy as np
 import torch
+from omegaconf.dictconfig import DictConfig
 
 from nemo.collections.nlp.data.language_modeling.megatron.base_dataset_utils import (
     get_datasets_weights_and_num_samples,
     get_train_valid_test_split_,
 )
 from nemo.collections.nlp.data.language_modeling.megatron.blendable_dataset import BlendableDataset
-from nemo.collections.nlp.data.language_modeling.megatron.gpt_dataset import (
-    _build_index_mappings,
-    get_indexed_dataset_,
-)
-from nemo.collections.nlp.data.language_modeling.megatron.indexed_retrieval_dataset import (
-    KNNIndex,
-    MMapRetrievalIndexedDataset,
-)
+from nemo.collections.nlp.data.language_modeling.megatron.indexed_dataset import deallocate_indexed_dataset_memory
+from nemo.collections.nlp.data.language_modeling.megatron.indexed_dataset import make_dataset as make_indexed_dataset
 from nemo.core import Dataset
 from nemo.utils import logging
 
 try:
-    from megatron.core import parallel_state
+    from megatron.core import mpu, tensor_parallel
+    from megatron.core.datasets.blended_megatron_dataset_builder import BlendedMegatronDatasetBuilder
+    from megatron.core.datasets.blended_megatron_dataset_config import GPTDatasetConfig
+    from megatron.core.datasets.gpt_dataset import GPTDataset
+    from megatron.core.models.retro.data.config import RetroGPTDatasets
+    from megatron.core.models.retro.data.query import get_retro_datasets
+    from megatron.core.models.retro.model import RetroConfig
+
+    from nemo.collections.nlp.modules.common.megatron.utils import get_ltor_masks_and_position_ids
 
     HAVE_MEGATRON_CORE = True
 
 except (ImportError, ModuleNotFoundError):
 
     HAVE_MEGATRON_CORE = False
 
-__all__ = [
-    "RETRODataset",
-    "build_train_valid_test_datasets",
-    "MockRETRODataset",
-    "build_mock_train_valid_test_datasets",
-]
```

Comment:
why delete this?

Reply:
I rewrote most of the retro_dataset.py to use mcore RETRO data-object. I find other models' datasets (e.g., gpt_dataset.py, bert_dataset.py) doesn't have this so just want to keep the format consistent.

Reply:
I have re-added the previous NeMo version files and changed them to `*_legacy`. I also have changed the import of relevant files to not break the code.